### PR TITLE
add pub details to pub card, and various css fixes

### LINKED ISSUE #420: Enrich publication representation
On the publications tab of each node page

- ~Add First Author last name, Title (First x words ok), Journal (abbreviations OK), Year~ #438 
- Add to the publications node page a hyperlinked list of mentioned diseases, phenotypes, etc. However, these should not be in a table as they might be construed as biological data instead of just context.

### PR DESCRIPTION
**Pull request by vincerubinetti**
related: #420 

- fixes footer css on small screens
- makes side nav bar fixed width for better readability, and collapses it sooner (as page width decreases) to avoid text wrapping clutter
- rewrites `getPublication` to accept list of ids instead of a single id, and refactors
- other minor css changes

still in progress

### CODE REVIEW

File: src/components/AssocTable.vue [hunk 1]
```diff
@@ -585,6 +598,9 @@ export default {
           assocSubject: subjectElem.label,
           subjectCurie: subjectElem.id,
           subjectLink,
+          title: truncate(elem.publicationMeta.title, 20),
```
Comment (falquaddoomi) [inline src/components/AssocTable.vue:601]: `publicationMeta` is sometimes undefined, causing this reference to `title` to throw an error.

Screen Shot 2021-07-30 at 10 09 39 AM

The error reliably occurs if you go to https://deploy-preview-438--monarch-ui.netlify.app/disease/MONDO:0007947, click "Publication" in the sidebar, then browse to page 4.

Comment (falquaddoomi) [reply src/components/AssocTable.vue:601]: Also, just an opinion, but 20 characters for a lot of these titles isn't much. I'd prefer to see more and possibly the option to expand the ellipses into the full title.

**Review by falquaddoomi — APPROVED** [review]
Looks good to me!

**Review by falquaddoomi — APPROVED** [review]

**Comment by vincerubinetti** [comment]
@kshefchek and/or @jmcmurry can you take a look at what I have so far (which addresses the first bullet point of #420). I didn't add publisher because the table was already getting too crowded, so I thought title, author, and year would be enough identifying information at a glance. And you can always click on the publication to get more authors and details.

**Comment by kshefchek** [comment]
perhaps instead of showing the PMID id we replace that with the title?  I think that's how it was originally intended but we ended up using PMIDs as a fall back (and as you can see we haven't stored many titles in our db)

**Comment by vincerubinetti** [comment]
> perhaps instead of showing the PMID id we replace that with the title? I think that's how it was originally intended but we ended up using PMIDs as a fall back (and as you can see we haven't stored many titles in our db)

This is implemented now.

**Comment by jmcmurry** [comment]
This looks amazing, thanks! My only feedback is that it would be great if possible to include the journal. To make room, we can jettison date specifics (Year is fine).

**Comment by vincerubinetti** [comment]
> This looks amazing, thanks! My only feedback is that it would be great if possible to include the journal. To make room, we can jettison date specifics (Year is fine).

Incorporated. I used the field `source` instead of `fulljournalname`, as `source` appears to consistently be an (accepted/standard?) abbreviation of the full journal name.

**Comment by vincerubinetti** [comment]
@jmcmurry Can you review the update and approve

**Review by jmcmurry — APPROVED** [review]
Amazing! Thanks

**Pull request merged**